Created Adding SSO Provider form

fkasapovic1 opened the pull request.

ekurtovic4 commented on frontend/src/components/AddSSOProviderForm.tsx:
```diff
@@ -0,0 +1,115 @@
+    };
+
+    try {
+      const response = await fetch("/sso-provider", {
```
neispravna ruta, submit forme ne funkcioniše ispravno

NedimBecic replied:
Nakon sto se promjeni da bude ispravna ruta, postoji greška sa json parsiranjem unesenih podataka

HarisMalisevic replied:
Greška s rutama je u kontorleru. Ruta koja je dogovorena je `api/sso-providers` ali kako je implementirano, ipravna ruta je `/api/sso-providers/sso-provider`

File `ssoProviders.routes.ts` prepraviti sa:
```ts
router.get("/sso-provider", SsoProviderController.getAllSSOProviders);
router.post("/sso-provider", SsoProviderController.addSSOProvider);
router.delete("/sso-provider/:id", SsoProviderController.deleteSSOProvider);
```
na:
```ts
router.get("/", SsoProviderController.getAllSSOProviders);
router.post("/", SsoProviderController.addSSOProvider);
router.delete("/:id", SsoProviderController.deleteSSOProvider);
```

Početni dio rute `/api/sso-providers` je definisan u `server.ts`

NedimBecic replied:
Vjerujem da je ovo popravljeno u pr-u za DocumentLayout i pushano na dev.

ekurtovic4 commented on frontend/src/components/AddSSOProviderForm.tsx:
```diff
@@ -0,0 +1,115 @@
+
+      if (response.ok) {
+        setResponseMessage("✅ SSO Provider successfully added!");
+        e.currentTarget.reset(); // Clear form
```
reset na ovaj način ne resetuje formu prilikom testiranja

ekurtovic4 reviewed (changes requested):
Pored komentara u kodu, sugestija je da se forma napravi manjom (u smislu uža forma, centrirana na stranici). Trenutna forma zauzima dosta prostora, a može se pretpostaviti da uneseni podaci neće biti toliko dugi. Možda pogledati već kreirane forme u okviru aplikacije.

NedimBecic reviewed (changes requested):
Generalno ispraviti greške koje je kolegica navela ranije i potencijalne novonastale greške prilikom tih promjena. Bilo bi dobro možda i malo opširnije testirati interakciju forme sa backendom prije slanje popravki.

HarisMalisevic commented on frontend/src/App.tsx:
```diff
@@ -24,6 +25,14 @@ function App() {
           }
         />
+        <Route
+          path="/sso-provider"
```
`path="/sso-provider"` nije konzistentno s drugim rutama
`path="/sso-providers"` (dodati 's' jer su sve slične rute u množini)

HarisMalisevic commented on frontend/src/components/AddSSOProviderForm.tsx:
```diff
@@ -0,0 +1,115 @@
+
+  return (
+    <div className="sso-form-container">
+      <div className="sso-form-wrapper">
```
Forma pokriva stranicu čitavom širinom. Klijent je to iskazao kao nepoželjno kod drugih projekata pa je bolje smanjiti na manju fiksnu širinu.

Pogledati formu na stranici `/document-types` kao uzor za imenzije.

fkasapovic1 commented:
Update:
Resolved conflicts 
Removed the comments.
Added a max-width to the form so it doesn’t stretch across the entire screen.
Updated the route to /api/sso-providers/sso-provider as specified.
Implemented a new method for resetting the form, which I tested locally.
Note:
Unfortunately, I was not able to test the backend side because login is not working on my end, most likely an issue on my side.
If you could switch to my branch and test it, or fix anything if needed. I dont think there are any major issues.

HarisMalisevic commented:
Kako bi dodavanje SSO Providera bilo univerzalno, u tabelu `sso_providers` sam dodao kolone:
```ts
authorization_url: {
    type: DataTypes.TEXT,
    allowNull: false,
},
token_url: {
    type: DataTypes.TEXT,
    allowNull: false,
},
```

Molim te @fkasapovic1 da dodaš ove dvije varijable u formu i da se one također šalju na istu rutu kao ranije varijable.

@NedimBecic, tebe ću zamoliti da u API rutama za dodavanje SSO providera dodaš ove dvije vrijednosti.

Branch `40-api-routes-for-login-with-custom-sso` sadrži potrebne izmjene modela pa to možete pull-ati.

Hvala unaprijed!
